lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
+      errorMessages.push('The provided shape type is not allowed.');
+      result = false;
+    }
 
-  if (allowedShapes.indexOf(shape.type) === -1) {
-    errorMessages.push('The provided shape type is not allowed.');
-    result = false;
-  }
+    if (shape.type === 'geometrycollection' && shape.coordinates) {
+      errorMessages.push('The coordinates property must not be provided for the "geometrycollection" shape type.');
+      result = false;
+    }
 
-  if (shape.type === 'geometrycollection' && shape.coordinates) {
-    errorMessages.push('The coordinates property must not be provided for the "geometrycollection" shape type.');
-    result = false;
-  }
+    if (shape.type !== 'geometrycollection' && (!shape.coordinates || !Array.isArray(shape.coordinates))) {
+      errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'geometrycollection' && (!shape.coordinates || !Array.isArray(shape.coordinates))) {
-    errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type === 'circle' && !shape.radius) {
+      errorMessages.push('The radius property is mandatory for the "circle" shape type.');
+      result = false;
+    }
 
-  if (shape.type === 'circle' && !shape.radius) {
-    errorMessages.push('The radius property is mandatory for the "circle" shape type.');
-    result = false;
-  }
+    if (shape.type !== 'circle' && shape.radius) {
+      errorMessages.push(`The radius property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'circle' && shape.radius) {
-    errorMessages.push(`The radius property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type !== 'polygon' && shape.type !== 'multipolygon' && shape.orientation) {
+      errorMessages.push(`The orientation property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'polygon' && shape.type !== 'multipolygon' && shape.orientation) {
-    errorMessages.push(`The orientation property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type !== 'geometrycollection' && shape.geometries) {
+      errorMessages.push(`The geometries property must not be provided for the "${shape.type}" shape type.`);
+      result = false;
+    }
 
-  if (shape.type !== 'geometrycollection' && shape.geometries) {
-    errorMessages.push(`The geometries property must not be provided for the "${shape.type}" shape type.`);
-    result = false;
-  }
+    if (shape.type === 'geometrycollection' && (!shape.geometries || !Array.isArray(shape.geometries))) {
+      errorMessages.push('The geometries property must be provided for the "geometrycollection" shape type.');
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 110 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)